feature_engine/transformation/date_transformer.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.base_transformers import BaseNumericalTransformer
```

Comment:
I don't think I would inherit this class. It has a check for numerical variables in the fit method that is not suitable here. And also, I would allow na, and make it optional for the user to say whether they want those to be raised or ignored.